### PR TITLE
New vpc channel resource support

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
The API web backend and policy need vpc channel configuration.

**Which issue this PR fixes**:
reference #1249 

**Special notes for your reviewer**:

**Release note**:

```release-note
1. new vpc channel resoruce support.
2. new acc test for each parameters.
3. new document.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed
test for ECS member type
```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccApigVpcChannelV2_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccApigVpcChannelV2_basic -timeout 360m -parallel 4
=== RUN   TestAccApigVpcChannelV2_basic
=== PAUSE TestAccApigVpcChannelV2_basic
=== CONT  TestAccApigVpcChannelV2_basic
--- PASS: TestAccApigVpcChannelV2_basic (546.90s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      547.017s
```
test for EIP member type
```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccApigVpcChannelV2_withEipMembers'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccApigVpcChannelV2_withEipMembers -timeout 360m -parallel 4
=== RUN   TestAccApigVpcChannelV2_withEipMembers
=== PAUSE TestAccApigVpcChannelV2_withEipMembers
=== CONT  TestAccApigVpcChannelV2_withEipMembers
--- PASS: TestAccApigVpcChannelV2_withEipMembers (540.80s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig540.868s
```

### CODE REVIEW

File: huaweicloud/services/apig/resource_huaweicloud_apig_vpc_channel.go [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+				Optional: true,
+			},
+			"members": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MaxItems: 10,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"ip_address": {
+							Type:     schema.TypeString,
```
Comment (ShiChangkuo) [inline huaweicloud/services/apig/resource_huaweicloud_apig_vpc_channel.go:139]: suggest to add **ExactlyOneOf** attribute

File: huaweicloud/services/apig/resource_huaweicloud_apig_vpc_channel.go [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+					return result, fmtp.Errorf("The instance ID is missing, please check your input of members")
+				}
+				info.EcsId = id.(string)
+				// Since the API requires that the name must be supported when the member type is 'ecs'.
+				// It is necessary to query the ecs instance information from server to obtain the instance name.
+				// We will cancel this unreasonable parameter configuration in the future.
+				ecsClient, err := config.ComputeV1Client(config.GetRegion(d))
+				if err != nil {
+					return result, fmtp.Errorf("Error creating HuaweiCloud ECS v1 client: %s", err)
+				}
```
Comment (ShiChangkuo) [inline huaweicloud/services/apig/resource_huaweicloud_apig_vpc_channel.go]: suggest to put those outside of the for loop.

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_vpc_channel_test.go [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+  instance_id = huaweicloud_apig_instance.test.id
+  port        = 8080
+  algorithm   = "WLC"
+  protocol    = "HTTPS"
+  path        = "/terraform/"
+  http_code   = "201,202,203"
+
+  members {
+    id = huaweicloud_compute_instance.test.id
+	weight     = 30
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_vpc_channel_test.go]: format error

File: docs/resources/apig_vpc_channel.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    weight = 70
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the APIG application resource.
```
Comment (ShiChangkuo) [inline docs/resources/apig_vpc_channel.md]: the description is wrong

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_vpc_channel_test.go [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+
+  bandwidth {
+    name        = "%s"
+    size        = 8
+    share_type  = "PER"
+    charge_mode = "traffic"
+  }
+}
+
+resource "huaweicloud_compute_eip_associate" "test" {
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_vpc_channel_test.go]: is it necessary to associate the eip to an ecs instance?

File: huaweicloud/services/apig/resource_huaweicloud_apig_vpc_channel.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_vpc_channel.go:139]: Negative, `ExactOneOf` cannot be used in `List` with a maximum length greater than 1. [link](https://github.com/hashicorp/terraform-plugin-sdk/issues/71)

Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_vpc_channel.go:139]: We can add related description to document.

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_vpc_channel_test.go (hunk 3 above)
Comment (Lance52259) [reply huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_vpc_channel_test.go]: Adjusted

File: huaweicloud/services/apig/resource_huaweicloud_apig_vpc_channel.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_vpc_channel.go]: done

File: docs/resources/apig_vpc_channel.md (hunk 4 above)
Comment (Lance52259) [reply docs/resources/apig_vpc_channel.md]: This description has been updated

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_vpc_channel_test.go (hunk 5 above)
Comment (Lance52259) [reply huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_vpc_channel_test.go]: thx, this is unnecessary and has been removed

**Pull request merged**